ainft/utils.py

```diff
@@ -35,6 +35,14 @@ def now() -> float:
     return datetime.now().timestamp()
 
 
+def truncate_text(text: str, max_length: int) -> str:
+    if max_length <= 0:
+        raise ValueError("max_length must be greater than 0.")
+    if len(text) > max_length:
+        return text[: max_length - 3] + "..."
```

Comment:
suggestion) how about adding some tests that this works well with unicode strings?

Reply:
@platfowner Thanks to your help, I found truncation failed with combined characters (e.g. é, ñ). So I now NFKD normalize first and then truncate text!